examples/demo13.py

```diff
@@ -0,0 +1,378 @@
+
+    # Make the 2-component parametric GSObjects for each object, including chromaticity (roughly
+    # appropriate SEDs per galaxy component, at the appropriate galaxy redshift).  Note that since
+    # the PSF is position-independent within the SCA, we can simply do the convolution with that PSF
+    # now instead of using a different one for each position.
+    gal_list = []
+    for i_gal in xrange(n_use):
+        logger.info('Processing the object at row %d in the input catalog.'%i_gal)
+
+        # Use built-in routines for constructing COSMOS objects:
+        gal = galsim.makeCOSMOSObj(cat, i_gal, chromatic=True)
+
+        # Convolve the chromatic galaxy and the chromatic PSF
+        final = galsim.Convolve(gal, PSF)
+        logger.debug('Pre-processing for galaxy %d completed.'%i_gal)
+        gal_list.append(final)
+
+    # Calculate the sky level for each filter, and draw the PSF and the galaxies through the
+    # filters.
+    for filter_name, filter_ in filters.iteritems():
+        logger.info('Beginning work for {0}.'.format(filter_name))
+
+        # Drawing PSF.  Note that the PSF object intrinsically has a flat SED, so if we
+        # convolve it with a galaxy, it will properly take on the SED of the galaxy.  However,
+        # this does mean that the PSF image being drawn here is not quite the right PSF for
+        # the galaxy.  Indeed, the PSF for the galaxy effectively varies within it, since it
+        # differs for the bulge and the disk.  To make a real image, one would have to choose SEDs
+        # for stars and convolve with a star that has a reasonable SED, but we are just going to
+        # draw with a flat SED for this demo.
+        out_filename = os.path.join(outpath, 'demo13_PSF_{0}.fits'.format(filter_name))
+        img_psf = galsim.ImageF(64,64)
+        PSF.drawImage(bandpass=filter_, image=img_psf, scale=wfirst.pixel_scale)
+        # Artificially normalize to a total flux of 1 for display purposes.
+        img_psf /= img_psf.array.sum()
+        img_psf.write(out_filename)
+        logger.debug('Created PSF with flat SED for {0}-band'.format(filter_name))
+
+        # Set up the full image that will contain all the individual galaxy images, with information
+        # about WCS:
+        final_image = galsim.ImageF(wfirst.n_pix,wfirst.n_pix, wcs=wcs)
+
+        # Draw the galaxies into the image.
+        for i_gal in xrange(n_use):
+            logger.info('Drawing image for the object at row %d in the input catalog'%i_gal)
+
+            # We want to only draw the galaxy once (for speed), not over and over with different
+            # sub-pixel offsets.  For this reason we will ignore the sub-pixel offset entirely.
+            stamp = galsim.Image(stamp_size, stamp_size)
+            gal_list[i_gal].drawImage(filter_, image=stamp)
+
+            # Have to find where to place it:
+            for i_gal_use in range(i_gal*n_per_gal, i_gal*(n_per_gal+1)):
```

Comment:
Oy, that's embarrassing.  Fixed.

To be honest, I am finding it hard to figure out if it looks unrealistic to me because it's actually something wrong with the galaxy population, or because it's so obvious that we have 10 galaxies repeated over and over.  The fact that one is a very elongated disk that appears with the exact same orientation 400 times is obviously unrealistic, but other than that, I think the only issue is that I had forgot to modify the sizes and fluxes to make it represent an I<25 sample.  As written it was representing an I<23 sample, which is a bit too bright for a density of 70/arcmin^2, but for an I<25 sample this isn't too crazy.  Can you see if the version I just pushed looks a bit better to you?  It does look a bit better to me though, again, there's still the "why are all the galaxies pointing the same way?" problem.  I think that one is unavoidable if we want to stick with just a few galaxies so the demo doesn't take forever.